h2o-core/src/main/java/water/api/RapidsHandler.java

```diff
@@ -59,18 +64,39 @@ public RapidsSchemaV3 exec(int version, RapidsSchemaV3 rapids) {
     }
   }
 
+  private static Class[] getRapidsClasses(){
```

Comment:
I was thinking of it. The problem is that SPI creates a new instance automatically, which we can't do with all rapids classes. I have written a bit different SPI (as part of some different project ) which doesn't create a new instance automatically and returns just loaded class, but that's just easy wrapper around this.

So after considering the above, the SPI doesn't seem to be a good fit here ( or maybe there are some tricks which I don't know :) )